x-pack/plugins/security_solution/public/attack_discovery/pages/session_storage/index.test.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { GenerationInterval } from '../../types';
+import {
+  encodeGenerationIntervals,
+  decodeGenerationIntervals,
+  getLocalStorageGenerationIntervals,
+  setLocalStorageGenerationIntervals,
+} from '.';
+
+const key = 'elasticAssistantDefault.attackDiscovery.default.generationIntervals';
+
+const generationIntervals: Record<string, GenerationInterval[]> = {
+  'test-connector-1': [
+    {
+      connectorId: 'test-connector-1',
+      date: new Date('2024-05-16T14:13:09.838Z'),
+      durationMs: 173648,
+    },
+    {
+      connectorId: 'test-connector-1',
+      date: new Date('2024-05-16T13:59:49.620Z'),
+      durationMs: 146605,
+    },
+    {
+      connectorId: 'test-connector-1',
+      date: new Date('2024-05-16T13:47:00.629Z'),
+      durationMs: 255163,
+    },
+  ],
+  testConnector2: [
+    {
+      connectorId: 'testConnector2',
+      date: new Date('2024-05-16T14:26:25.273Z'),
+      durationMs: 130447,
+    },
+  ],
+  testConnector3: [
+    {
+      connectorId: 'testConnector3',
+      date: new Date('2024-05-16T14:36:53.171Z'),
+      durationMs: 46614,
+    },
+    {
+      connectorId: 'testConnector3',
+      date: new Date('2024-05-16T14:27:17.187Z'),
+      durationMs: 44129,
+    },
+  ],
+};
+
+describe('storage', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('encodeGenerationIntervals', () => {
+    it('returns null when generationIntervals is invalid', () => {
+      const invalidGenerationIntervals: Record<string, GenerationInterval[]> =
+        1n as unknown as Record<string, GenerationInterval[]>; // <-- invalid
```

Comment:
That's a `BigInt`! Nice way of exercising `JSON.stringify()` here, didn't know about this one 🙂